src/containers/Tenant/Diagnostics/Partitions/Partitions.tsx

```diff
@@ -43,215 +40,162 @@ export const b = block('ydb-diagnostics-partitions');
+            setColumnsIdsForSelector(generalPartitionColumnsIds);
+        }
+    }, [selectedConsumer, partitionsWithConsumers, partitionsWithoutConsumers, nodesMap]);
+
+    const fetchData = useCallback(
         (isBackground: boolean) => {
-            if (!isBackground) {
-                dispatch(setDataWasNotLoaded());
+            if (!selectedConsumer) {
+                if (!isBackground) {
+                    dispatch(setTopicDataWasNotLoaded());
+                }
+                dispatch(getTopic(componentCurrentPath));
             }
 
             if (selectedConsumer && consumers && consumers.includes(selectedConsumer)) {
+                if (!isBackground) {
+                    dispatch(setConsumerDataWasNotLoaded());
+                }
+
                 dispatch(getConsumer(componentCurrentPath, selectedConsumer));
             }
         },
         [dispatch, selectedConsumer, componentCurrentPath, consumers],
     );
 
-    useAutofetcher(fetchConsumerData, [fetchConsumerData], autorefresh);
-
-    const consumersToSelect = useMemo(
-        () =>
-            consumers
-                ? consumers.map((consumer) => ({
-                      value: consumer,
-                      content: consumer,
-                  }))
-                : undefined,
-        [consumers],
-    );
+    useAutofetcher(fetchData, [fetchData], autorefresh);
 
+    // Wrong consumer could be passed in search query
+    // Reset consumer if it doesn't exist for current topic
     useEffect(() => {
-        const shouldUpdateSelectedConsumer =
-            !selectedConsumer || (consumers && !consumers.includes(selectedConsumer));
+        const isTopicWithoutConsumers = topicWasLoaded && !consumers;
+        const wrongSelectedConsumer =
+            selectedConsumer && consumers && !consumers.includes(selectedConsumer);
 
-        if (consumersToSelect && consumersToSelect.length && shouldUpdateSelectedConsumer) {
-            dispatch(setSelectedConsumer(consumersToSelect[0].value));
+        if (isTopicWithoutConsumers || wrongSelectedConsumer) {
+            dispatch(setSelectedConsumer());
         }
-    }, [dispatch, consumersToSelect, selectedConsumer, consumers]);
-
-    const selectedColumns: string[] = useMemo(
-        () =>
-            savedSelectedColumns
-                ? JSON.parse(savedSelectedColumns)
-                : PARTITIONS_DEFAULT_SELECTED_COLUMNS,
-        [savedSelectedColumns],
-    );
-
-    const columnsToSelect = useMemo(() => {
-        return Object.values(PARTITIONS_COLUMNS_IDS).map((id) => {
-            return {
-                title: PARTITIONS_COLUMNS_TITILES[id],
-                selected: Boolean(selectedColumns?.includes(id)),
-                id: id,
-                required: id === PARTITIONS_COLUMNS_IDS.PARTITION_ID,
-            };
-        });
-    }, [selectedColumns]);
+    }, [dispatch, topicWasLoaded, selectedConsumer, consumers]);
 
     const columnsToShow = useMemo(() => {
```

Comment:
This may be moved into custom hook `useVisibleColumns = (initialColumns, hiddenColumns) => {return [visibleColumns, setColumns]}`
`hiddenColumns` - may be even key for LS, if it is common to store hidden columns for different tables in LS.

Reply:
I’m not sure that there is much use in moving columns into a custom hook. However, I created a hook `useSetting` to simplify interaction with stored settings